internal/requests/requests.go

```diff
@@ -10,45 +11,101 @@ import (
+	// We don't validate the specific type as it's dynamic and can only be done
+	// server side.
+	Type                 string `json:"type" yaml:"type" valid:"required~type is required"`
+	WorkflowTemplateName string `json:"workflow_template_name" yaml:"workflow_template_name" valid:"required~workflow_template_name is required"`
 }
 
-// ValidateFramework is an optional validation that should be passed as parameter to Validate().
-func (req CreateWorkflow) ValidateFramework(frameworks []string) func() error {
-	return func() error {
-		return validations.ValidateVar("framework", req.Framework, fmt.Sprintf("oneof=%s", strings.Join(frameworks, " ")))
+// Validate validates CreateWorkflow.
+func (req CreateWorkflow) Validate(optionalValidations ...func() error) error {
+	v := []func() error{
+		func() error { return validations.ValidateStruct(req) },
+		req.validateArguments,
+		req.validateParameters,
 	}
+	v = append(v, optionalValidations...)
+
+	return validations.Validate(v...)
 }
 
 // ValidateType is an optional validation should be passed as parameter to Validate().
 func (req CreateWorkflow) ValidateType(types []string) func() error {
 	return func() error {
-		return validations.ValidateVar("type", req.Type, fmt.Sprintf("oneof=%s", strings.Join(types, " ")))
+		for _, t := range types {
+			if req.Type == t {
+				return nil
+			}
+		}
+
+		return fmt.Errorf("type must be one of '%s'", strings.Join(types, " "))
 	}
 }
 
-// Validate validates CreateWorkflow.
-func (req CreateWorkflow) Validate(optionalValidations ...func() error) error {
-	for _, validation := range optionalValidations {
-		if err := validation(); err != nil {
-			return err
+// validateParameters validates the Parameters.
+// 'execute_container_image_uri' is required and the URI format will be
+// validated.
+// 'pre_container_image_uri' is optional. If it's provided, the URI format will
+// be validated.
+func (req CreateWorkflow) validateParameters() error {
+	val, ok := req.Parameters["execute_container_image_uri"]
+	if !ok {
+		return errors.New("parameter execute_container_image_uri is required")
+	}
+
+	if !validations.IsValidImageURI(val) {
+		return errors.New("parameter execute_container_image_uri must be a valid container uri")
+	}
+
+	if val, ok := req.Parameters["pre_container_image_uri"]; ok {
+		if !validations.IsValidImageURI(val) {
+			return errors.New("parameter pre_container_image_uri must be a valid container uri")
 		}
 	}
-	return validations.ValidateStruct(req)
+
+	return nil
+}
+
+// validateArguments validates the Arguments.
+// The valid Arguments cases are:
+// * no arguments
+// * both 'execute' and 'init'
+// TODO long term, we should evaluate if hard coding in code is the right
+// approach to specifying different argument types vs allowing dynamic
+// specification and interpolation in service/config.yaml
+func (req CreateWorkflow) validateArguments() error {
```

Comment:
It really wasn't clear what [the previous code](https://github.com/argoproj-labs/argo-cloudops/blob/3db7dac356db59bd0e5a08fd96fa009c23d5735b/internal/validations/validations.go#L93) was trying to validate prior. I _think_ this is what it was trying to do.